File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func hasCycle(head *ListNode) bool {
    runner1 := head
    runner2 := head


    for runner2 != nil && runner2.Next != nil {
        runner1 = runner1.Next
        runner2 = runner2.Next.Next

        if runner1 == runner2 {
            return true
        }
    }


    return false
    
}

File: Data Structures & Algorithms/palindrome-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        arr = []
        curr = head


        while curr:
            arr.append(curr.val)
            curr = curr.next
        

        i , j = 0 , len(arr) - 1

        while i < j :
            if arr[i] != arr[j]:
                return False
            i += 1
            j -= 1
        
        return True